src/main/java/uk/gov/hmcts/reform/idam/health/vault/AccessTokenKeyVaultCredential.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.idam.health.vault;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.credentials.AzureTokenCredentials;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.DefaultHttpRequestRetryHandler;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class AccessTokenKeyVaultCredential extends AzureTokenCredentials {
+
+    private static final String MSI_URL = "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fvault.azure.net";
+
+    private final static String METADATA_HEADER = "Metadata";
+    private final static String ACCESS_TOKEN_KEY = "access_token";
+
+    private final String tokenEndpoint;
+
+    public AccessTokenKeyVaultCredential() {
+        this(MSI_URL);
+    }
+
+    public AccessTokenKeyVaultCredential(String tokenEndpoint) {
```

Comment:
If it exists to help with tests, maybe we could make it `protected`?

Reply:
Yes, although I was also still tempted to move the token endpoint into the properties file. Partly because I think it belongs there, but also It might make it easier to run the health check locally with a fake endpoint. It's very unlikely this endpoint will ever change though, so maybe it's slightly more hassle than it's worth (although a property for the MSI token endpoint would be an obvious replacement for the "credential type" switch)

Reply:
Moved to make easier to test with fake endpoint.